### PR TITLE
fix #16364

### LINKED ISSUE #16364: `&` shows as `&amp;` in docs
### Example
in https://nim-lang.github.io/Nim/lib.html search for `&`

### Current Output

image

### Expected Output
show `&`, as it did in stable docs 1.4.x

image

### Additional Information
devel 1.5.1 e843492b1334ded1d8fcd0e0eb6dc94fcf970aba
there could be more such symbols

### PR DESCRIPTION
**Pull request by ringabout**
image

### CODE REVIEW

File: tools/dochack/dochack.nim [hunk 1]
```diff
@@ -268,15 +268,13 @@ var
 template normalize(x: cstring): cstring = x.toLower.replace("_", "")
 
 proc escapeCString(x: var cstring) =
+  # Original strings are already esacped except HTML tags, so
```
Comment (Clyybber) [inline tools/dochack/dochack.nim]:
```suggestion
  # Original strings are already escaped except HTML tags, so
```

**Comment by timotheecour** [comment]
- [ ] still broken for `//`

http://localhost:7029/system.html#%3C//%3E.t%2Cuntyped

image

https://nim-lang.github.io/Nim/lib.html is "less broken":

image



should show like this:

image

given that it shows correctly in last screenshot, can we reuse the escaping code used to generate proc declarations, since it seems to work?


## note
hard to test search box functionality but maybe add something like this to nimdoc/testproject/testproject.nim so we can at least inspect manually that search works 
```nim
proc `bar`*() = discard
proc bar2*() = discard
proc bar3*() = discard
proc `\\`*() = discard
proc `\/`*() = discard
proc `/\`*() = discard
proc `++`*() = discard
proc `&&`*() = discard
proc `||`*() = discard
proc `^`*() = discard
proc `|`*() = discard
proc `->`*() = discard
proc `--`*() = discard
proc `&=`*() = discard
proc `//`*() = discard
proc `<//>`*() = discard
proc `</\>`*() = discard
proc `<\\>`*() = discard
```

**Review by timotheecour — APPROVED** [review]
LGTM since it improves current situation without making anything worse, but if you can fix https://github.com/nim-lang/Nim/pull/16379#issuecomment-747619505 that would be nice

> given that it shows correctly in last screenshot, can we reuse the escaping code used to generate proc declarations, since it seems to work?

**Comment by ringabout** [comment]
It's not easy to fix that I guess, it is broken in JS parts.
```nim
    echo matches[i][0].innerHTML
    escapeCString(matches[i][0].innerHTML)
    echo matches[i][0].innerHTML
```
Output:
```
dochack.js:993 system: `<!--/-->`(t: untyped): untyped
dochack.js:993 system: `&lt;!--/--&gt;`(t: untyped): untyped
```
If only `<//>` is broken, let's leave it broken.

**Pull request merged**